src/vm/costs/mod.rs

```diff
@@ -413,21 +419,30 @@ fn load_cost_functions(clarity_db: &mut ClarityDatabase) -> Result<CostStateSumm
                     );
                     continue;
                 }
-                if !c.public_function_types.contains_key(&cost_function)
-                    && !c.read_only_function_types.contains_key(&cost_function)
-                    && !c.private_function_types.contains_key(&cost_function)
-                {
+
+                let cost_function_types =
+                    c.public_function_types.get(&cost_function).or_else(|| {
+                        c.read_only_function_types
+                            .get(&cost_function)
+                            .or_else(|| c.private_function_types.get(&cost_function).clone())
+                    });
+
+                if cost_function_types.is_none() {
                     warn!("Confirmed cost proposal invalid: cost-function-name not defined";
                           "confirmed_proposal_id" => confirmed_proposal,
                           "contract_name" => %cost_contract,
                           "function_name" => %cost_function,
                     );
                     continue;
                 }
```

Comment:
For these gets, the `if let` construct allows you to do this without an unwrap.

```suggestion
                let cost_function_opt =  c.read_only_function_types
                            .get(&cost_function)
                            .or_else(|| c.private_function_types.get(&cost_function))
                    });
                let cost_function_types = if let Some(result) = cost_function_opt {
                    result
                } else {
                    warn!("Confirmed cost proposal invalid: cost-function-name not defined";
                          "confirmed_proposal_id" => confirmed_proposal,
                          "contract_name" => %cost_contract,
                          "function_name" => %cost_function,
                    );
                    continue;
                }
                
                
                (
                    ClarityCostFunctionReference {
                        contract_id: cost_contract,
                        function_name: cost_function.to_string(),
                    },
                    cost_function_types.clone(),
                )
```

(Also, we don't need to check for public functions here, because the cost function couldn't be a public function anyways).